src/mappings/token.ts

```diff
@@ -32,6 +35,59 @@ async function isCompleteMint(
     return !!(await ctx.store.get(Mint, mintId))?.sender // sufficient checks
 }
 
+function beautifyCurrencyIdString(event: any) {
+    let currencyId = ''
+    switch (event.currencyId.__kind) {
+        case 'ZenlinkLPToken': {
+            currencyId =
+                'ZenlinkLPToken(' + String(event.currencyId.value) + ')'
+            break
+        }
+        case 'Native': {
+            currencyId = 'Native'
+            break
+        }
+        case 'Stellar': {
+            switch (event.currencyId.value.__kind) {
+                case 'StellarNative': {
+                    currencyId = 'StellarNative'
+                    break
+                }
+                case 'AlphaNum4': {
+                    currencyId =
+                        'Stellar::AlphaNum4(' +
+                        String(event.currencyId.value.code) +
```

Comment:
Oh and can we also have the `issuer` field here and for `AlphaNum12`?
Something like 
```
Stellar::AlphaNum4(' +
                        String(event.currencyId.value.code) 
                        + "," 
                        + String(event.currencyId.value.issuer)
                        ')'
```
should work probably (?)

Reply:
@ebma, thanks for the comment.

I did it initially, but the issuer information is represented as bytes. I'll attempt to convert between bytes and SS58 addresses, as we've done in the past.

Reply:
Oh... true I did not think about that. Let's not use SS58 here though because Stellar is not using that and it would spit out some info that is not useful in that context. Instead we should try to derive the Stellar public key from that raw bytes.

I think we should be able to use [this](https://github.com/stellar/js-stellar-base/blob/5ed52ab68d6c5305a10f4532087e08de75c814c5/src/address.js#L46) function to get the public key from the raw binary data. So we need to:
- add the `stellar-base` dependency (`npm install stellar-base`)
- import the `Address` type from `stellar-base` 
- use `Address.account(event.currencyId.value.issuer).toString()` to print the public key

I would suggest that we move this logic to derive the public key from the issuer to an extra function again. Something like
```
function deriveStellarPublicKeyFromRaw(raw: <type>) {
   const address = Address.account(event.currencyId.value.issuer);
   return address.toString()
}
```

Reply:
Done! :muscle: